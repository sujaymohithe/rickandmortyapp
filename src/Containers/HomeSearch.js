import React, { useEffect } from 'react';
import { Container } from 'react-bootstrap';
import { connect } from 'react-redux';
import * as actions from '../store/actions/actions';
import CharacterList from '../Components/CharacterList/CharacterList';
import './HomeSearch.css';

const HomeSearch = props => {
    const {
        characters,
        characterDetails,
        getCharactersList,
        getCharacterDetailsById,
        getOtherRecommendationList } = props;

    useEffect(() => {
        getCharactersList();
    }, [getCharactersList]);

    return (
        <Container className="">
            <CharacterList
                characters={characters}
                characterDetails={characterDetails}
                getCharacterList={getCharactersList}
                getCharacterDetailsById={getCharacterDetailsById}
                getOtherRecommendationList={getOtherRecommendationList}
            />
        </Container>
    );
}

//method that copies part of the state to the props of this component.
export const mapStateToProps = state => {
    return {
        characters: state.charactersList,
        characterDetails: state.characterDetails,
    };
};

//these functions will be accessible via props in child components
export const mapDispatchToProps = dispatch => {
    return {
        getCharactersList: (url, filterType, filterValue) =>
            dispatch(actions.getCharactersList(url, filterType, filterValue)),
        getCharacterDetailsById: (id) =>
            dispatch(actions.getCharacterDetailsById(id)),
        getOtherRecommendationList: (url, filterType, filterValue) =>
            dispatch(actions.getOtherRecommendationList(url, filterType, filterValue)),
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(HomeSearch);
