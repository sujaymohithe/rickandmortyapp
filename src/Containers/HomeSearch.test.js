import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import { store } from '../store/store';
import HomeSearch from './HomeSearch';

Enzyme.configure({ adapter: new Adapter() });

describe('<HomeSearch/> Container', () => {
    const container = shallow(
        <Provider store={store}>
            <HomeSearch />
        </Provider>
    );

    it('should render the HomeSearch container', () => {
        expect(container.length).toEqual(1);
        expect(container).toMatchSnapshot();
    });

});