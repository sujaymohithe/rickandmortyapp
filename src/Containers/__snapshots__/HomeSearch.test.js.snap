// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeSearch/> Container should render the HomeSearch container 1`] = `ShallowWrapper {}`;
